actions/format_image_action.go

```diff
@@ -0,0 +1,157 @@
+/*
+FormatImage Action
+
+This action creates an image file and formats it with a filesystem.
+
+Yaml syntax:
+ - action: format-image
+   imagename: image_name
+   imagesize: size
+   fs: filesystem
+   blocksize: 4096
+
+Mandatory properties:
+
+- imagename -- the name of the image file.
+
+- imagesize -- generated image size in human-readable form, examples: 100MB, 1GB, etc.
+
+- fs -- filesystem type used for formatting.
+
+Optional properties:
+
+- blocksize -- size of blocks in bytes, valid values are 1024, 2048 and 4096. For ext fs only.
+*/
+
+package actions
+
+import (
+	"fmt"
+	"github.com/docker/go-units"
+	"github.com/go-debos/fakemachine"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+
+	"github.com/go-debos/debos"
+)
+
+type FormatImageAction struct {
+	debos.BaseAction `yaml:",inline"`
+	ImageName        string
+	ImageSize        string
+	FS               string
+	BlockSize        int
+	size             int64
+}
+
+func (i FormatImageAction) format(context debos.DebosContext) error {
+	label := fmt.Sprintf("Formatting image")
+
+	cmdline := []string{}
+	switch i.FS {
+	case "vfat":
+		cmdline = append(cmdline, "mkfs.vfat")
+	case "btrfs":
+		// Force formatting to prevent failure in case if partition was formatted already
+		cmdline = append(cmdline, "mkfs.btrfs", "-f")
+	case "none":
+	default:
+		cmdline = append(cmdline, fmt.Sprintf("mkfs.%s", i.FS))
+	}
+
+	if len(cmdline) != 0 {
+		cmdline = append(cmdline, context.Image)
+
+		cmd := debos.Command{}
+		if err := cmd.Run(label, cmdline...); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (i FormatImageAction) PreMachine(context *debos.DebosContext, m *fakemachine.Machine,
+	args *[]string) error {
+	image, err := m.CreateImage(i.ImageName, i.size)
+	if err != nil {
+		return err
+	}
+
+	context.Image = image
+	*args = append(*args, "--internal-image", image)
+	return nil
+}
+
+func (i *FormatImageAction) PreNoMachine(context *debos.DebosContext) error {
+	// XXXX Couldn't test this function.
```

Comment:
Don't add todo's like this.. PreNoMachine is when you run it under sudo without fakemachine support (e.g. no /dev/kvm support(..

That said probably worth refactoring this fout of image-partition again rather then copying it here